experimental/PropertyDDS/packages/property-binder/src/data_binder/dataBinder.ts

```diff
@@ -0,0 +1,2824 @@
+                throw error;
+            }
+
+            this.popBindingActivationScope();
+        }
+
+        if (this._activationScope === 0) {
+            this._postChangesetProcessing();
+        }
+    }
+
+    /**
+     * Set/increment a letiable on the data binding that says that it is registered with one or more
+     * DataBinders. This allows DataBinding.registerOnPath to alert the user when they are registering a
+     * path or property to a DataBinding after it is already registered in the manager.
+     *
+     * @param in_bindingConstructor - the constructor to modify
+     * @hidden
+     */
+    private _markDataBindingAsRegistered(in_bindingConstructor: typeof DataBinding) {
+        if (in_bindingConstructor.prototype.hasOwnProperty('__numDataBinders')) {
+            in_bindingConstructor.prototype.__numDataBinders!++;
+        } else {
+            in_bindingConstructor.prototype.__numDataBinders = 1;
+        }
+    }
+
+    /**
+     * Clear/decrement a letiable on the data binding that says that it is registered with one or more
+     * DataBinders. This allows DataBinding.registerOnPath to alert the user when they are registering a
+     * path or property to a DataBinding after it is already registered in the manager.
+     *
+     * @param in_dataBindingConstructor - the constructor to modify
+     *
+     * @hidden
+     */
+    private _unmarkDataBindingAsRegistered(in_dataBindingConstructor: typeof DataBinding) {
+        const hasCount = in_dataBindingConstructor.prototype.hasOwnProperty('__numDataBinders');
+        console.assert(hasCount && in_dataBindingConstructor.prototype.__numDataBinders! > 0);
+        if (hasCount) {
+            in_dataBindingConstructor.prototype.__numDataBinders!--;
+            console.assert(in_dataBindingConstructor.prototype.__numDataBinders! >= 0);
+            if (in_dataBindingConstructor.prototype.__numDataBinders === 0) {
+                // Might as well clean up
+                delete in_dataBindingConstructor.prototype.__numDataBinders;
+            }
+        }
+    }
+
+    /**
+     * Create a traversal context as if we were traversing the hierarchy at this location
+     *
+     * @param in_property - the property at the given path
+     * @param in_traversalPath - the absolute path as a string
+     * @param in_absTokenizedPath - the absolute path, tokenized
+     * @param in_dataBindingTreeNode - the DataBindingTree node at the given path
+     *
+     * @returns a fake traversal context
+     *
+     * @hidden
+     */
+    private _createFakeTraversalContext(in_property: BaseProperty, in_traversalPath: string, in_absTokenizedPath: Array<string>,
+        in_dataBindingTreeNode: NodeType): Utils.TraversalContext {
+        let fakeContext = new TraversalContext({
+            fullPostPath: in_traversalPath,
+            fullPath: in_traversalPath,
+            propertyContainerType: in_property.getContext() as any,
```

Comment:
`getContext` returns `string`. Maybe TraversalContext options should be updated, so we don't have to convert to any?

Reply:
Yes, I was planning to do that, but then it's a bigger effort. I would prefer to to keep these improvements in separated PR. Otherwise, this PR will get even much bigger than it is now.